### ISSUE TITLE
yield in comprehension check should allow the outermost comprehension

### ISSUE THREAD

**Issue by JelleZijlstra** [microsoft/pyright#2871]
```python
def capybara(y):
    [x for x in (yield y)]  # pyright: "yield" not allowed inside a list comprehension
```
As of https://bugs.python.org/issue10544, yielding inside a comprehension is deprecated, but the outermost iterable is evaluated in the enclosing scope, not in the comprehension's scope, and yielding there is fine.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

It occurs to me that we need more code samples in the world that use `capybara` as an identifier name. :)

**Comment by JelleZijlstra** [comment]
I got into the habit of naming my test functions "capybara" for some reason :) I am finding these bugs by looking at the pyanalyze test suite in VSCode and it's full of capybaras, so you might see some more.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.210, which I just published. It will also be included in the next release of pylance.
